Fix panel header styles

adekbadek opened the pull request:
## Description

af3153afe99f9c4be614c2943a6ad1559c846534 & 3ca05a7d0ef966df724ebe744b04efd4330e7c20 have effectively updated the color of any `.components-button` button placed in the side panel. 
This selector should however affect only the button in the panel title. Otherwise, it also affects any `.components-button` placed in the side panel.

## How has this been tested?

by changing the selector in Chrome devtools

## Screenshots <!-- if applicable -->

before: 

![image](https://user-images.githubusercontent.com/7383192/72982951-8c0f4100-3de0-11ea-98ec-2b23917b002b.png)

after:

![image](https://user-images.githubusercontent.com/7383192/72982995-a47f5b80-3de0-11ea-9aab-09665abb6288.png)
## Types of changes
<!-- What types of changes does your code introduce?  -->
<!-- Bug fix (non-breaking change which fixes an issue) -->
<!-- New feature (non-breaking change which adds functionality) -->
<!-- Breaking change (fix or feature that would cause existing functionality to not work as expected) -->

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [ ] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [ ] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [ ] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR. <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/master/docs/contributors/native-mobile.md -->.

adekbadek committed:
Fix panel header styles

This selector should affect only the button in the panel title. Otherwise, it also affects any .components-button placed in the side panel.

adekbadek committed:
Remove EOL spaces

adekbadek commented:
It looks like the CI failed for an unrelated reason: https://travis-ci.com/WordPress/gutenberg/jobs/279276806 
A rerun should fix that

ItsJonQ commented:
@adekbadek Hi there 👋 

Thank you for catching this one :). I took a look and was able to replicate the issue and can confirm that your update fixes it.

In taking a look at the `panel/style.scss` file, I noticed that there were  `<Button />` styles defined later on the file. It specifically targeted the `<Button />` component that renders within a `Panel Body Title`.

https://github.com/WordPress/gutenberg/blob/f41385118905fc36206b34ab8325e1a820cb95bb/packages/components/src/panel/style.scss#L89

I did a quick test by removing the CSS rules from the `.components-panel__body` entirely.

```diff
-	// panel section header
-	&-title > .components-button {
-		color: $dark-gray-900;
-	}
```

It looked like that did it!

@adekbadek Does the change I proposed above make sense? 

Thanks again!

adekbadek committed:
Remove CSS rule for .components-button

Styling for the Button in the panel is provided later on the file:
https://github.com/WordPress/gutenberg/blob/f41385118905fc36206b34ab8325e1a820cb95bb/packages/components/src/panel/style.scss#L89

adekbadek commented:
@ItsJonQ That's true, I didn't check that. So the rule can be removed altogether: https://github.com/adekbadek/gutenberg/commit/61c5b69ab2cfdaa3629eac4df8592ae1b8fb27ca

ItsJonQ reviewed (approved):
🚀 from me! The changes look good. The code works as expected when testing locally

youknowriad commented:
Good luck for your first contribution.
